ci/manual_release.md

```diff
@@ -1,3 +1,144 @@
+# Creating a new Knative release locally
+
+**Note:** This should not be used for official releases.
+
+This is intended for debugging and testing purposes.
+
+## The release.sh script
+
+The `release.sh` script automates the creation of Knative releases, either nightly
+or versioned ones.
+
+By default, the script creates a nightly release but does not publish it
+anywhere.
+
+## Common flags for cutting releases
+
+The following flags affect the behavior of the script, no matter the type of the
+release.
+
+- `--skip-tests` Do not run tests before building the release. Otherwise, build,
+  unit and end-to-end tests are run and they all must pass for the release to be
+  built.
+- `--tag-release`, `--notag-release` Tag (or not) the generated images with
+  either `vYYYYMMDD-<commit_short_hash>` (for nightly releases) or `vX.Y.Z` for
+  versioned releases. _For versioned releases, a tag is always added._
+- `--release-gcs` Defines the GCS bucket where the manifests will be stored. By
+  default, this is `knative-nightly/MODULE` (MODULE is a Knative module name, e.g.
+  `serving` or `eventing`). This flag is ignored if the release is not being published.
+- `--release-gcr` Defines the GCR where the images will be stored. By default,
+  this is `gcr.io/knative-nightly`. This flag is ignored if the release is not
+  being published.
+- `--publish`, `--nopublish` Whether the generated images should be published to
+  a GCR, and the generated manifests written to a GCS bucket or not. If yes, the
+  `--release-gcs` and `--release-gcr` flags can be used to change the default
+  values of the GCR/GCS used. If no, the images will be pushed to the `ko.local`
+  registry, and the manifests written to the local disk only (in the repository
+  root directory).
+
+## Creating nightly releases
+
+Nightly releases are built against the current git tree. The behavior of the
+script is defined by the common flags. You must have write access to the GCR and
+GCS bucket the release will be pushed to, unless `--nopublish` is used.
+
+Examples:
+
+```bash
+# Create and publish a nightly, tagged release.
+./hack/release.sh --publish --tag-release
+
+# Create release, but don't test, publish or tag it.
+./hack/release.sh --skip-tests --nopublish --notag-release
+```
+
+## Creating versioned releases
+
+To specify a versioned release to be cut, you must use the `--version` flag.
+Versioned releases are usually built against a branch in the Knative
+repository, specified by the `--branch` flag.
+
+- `--version` Defines the version of the release, and must be in the form
+  `X.Y.Z`, where X, Y and Z are numbers.
+- `--branch` Defines the branch in Knative repository from which the
+  release will be built. If not passed, the `master` branch at HEAD will be
+  used. This branch must be created before the script is executed, and must be
+  in the form `release-X.Y`, where X and Y must match the numbers used in the
```

Comment:
if it must be `release-X.Y`, would it fail if branch not passed, since it's using `master`

Reply:
No. It only fails if you use the flag and pass a badly named branch.